result.go

```diff
@@ -381,17 +381,42 @@ func (q Query) ToMapBy(
 	res.Elem().Set(m)
 }
 
-// ToSlice iterates over a collection and populates the result slice with elements.
+// ToSlice iterates over a collection, copy to result slice the collection elements,
```

Comment:
Let's fix this godoc. It's confusing to me, as it says:

1. "iterates over a collection" ––do we mean "slice"? what are some other possible collections here?
2. "copy to result slice the collection elements" ––sounds like a grammatical mistake here?
3. "appends items if initial capacity is not enough" ––probably should sound more like: _"If it has sufficient capacity, the destination is resliced to accommodate the new elements. If it does not, a new underlying array will be allocated."_ [[source](https://golang.org/pkg/builtin/#append)]
4. we should still indicate that the slice is resliced to reflect the size of the result.